cdap-app-fabric/src/test/java/co/cask/cdap/gateway/handlers/OperationsDashboardHttpHandlerTest.java

```diff
@@ -52,61 +62,153 @@
+        ImmutableMap.of(SCHEDULED_PROG2_ID.getProgram(), scheduledWorfklow2), Collections.EMPTY_MAP,
+        Collections.EMPTY_MAP, Collections.EMPTY_MAP, Collections.EMPTY_MAP);
+    store.addApplication(APP2_ID, dummyAppSpec2);
+    // add a schedule to be triggered every 5 minutes for SCHEDULED_PROG1_ID
+    int sched1Mins = 5;
+    ProgramSchedule sched1 = new ProgramSchedule("tsched1", "five min schedule", SCHEDULED_PROG1_ID,
+      Collections.EMPTY_MAP, new TimeTrigger(String.format("*/%d * * * *", sched1Mins)), Collections.emptyList());
+    scheduler.addSchedule(sched1);
+    scheduler.enableSchedule(sched1.getScheduleId());
+    // add a schedule to be triggered every 10 minutes for SCHEDULED_PROG2_ID
+    int sched2Mins = 10;
+    ProgramSchedule sched2 = new ProgramSchedule("tsched2", "ten min schedule", SCHEDULED_PROG2_ID,
+      Collections.EMPTY_MAP, new TimeTrigger(String.format("*/%d * * * *", sched2Mins)), Collections.emptyList());
+    scheduler.addSchedule(sched2);
+    scheduler.enableSchedule(sched2.getScheduleId());
+    long currentTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis());
+    // the duration of the query time range
+    int durationSecs = 3600;
+    // get ops dashboard results between current time and current time + 3600 from TEST_NAMESPACE1 and TEST_NAMESPACE2
+    String opsDashboardQueryPath =
+      String.format("%s/dashboard?start=%d&duration=%d&namespace=%s&namespace=%s",
+        BASE_PATH, currentTime, durationSecs, TEST_NAMESPACE1, TEST_NAMESPACE2);
+    List<DashboardProgramRunRecord> dashboardRecords = getDashboardRecords(opsDashboardQueryPath);
+    // calculate the expected number of scheduled runs according to the query duration and the schedules
+    long expectedScheduledProgram1 = durationSecs / TimeUnit.MINUTES.toSeconds(sched1Mins);
+    long expectedScheduledProgram2 = durationSecs / TimeUnit.MINUTES.toSeconds(sched2Mins);
+    // assert the number of scheduled runs are expected for both programs
+    Assert.assertEquals(expectedScheduledProgram1,
+      dashboardRecords.stream().filter(record -> SCHEDULED_PROG1_ID.getProgram().equals(record.getProgram())).count());
+    Assert.assertEquals(expectedScheduledProgram2,
+      dashboardRecords.stream().filter(record -> SCHEDULED_PROG2_ID.getProgram().equals(record.getProgram())).count());
+    // assert the artifact id is correct
+    Assert.assertTrue(dashboardRecords.stream()
+                        .filter(record -> SCHEDULED_PROG1_ID.getProgram().equals(record.getProgram()))
+                        .allMatch(schedule -> ArtifactSummary.from(ARTIFACT1_ID).equals(schedule.getArtifact())));
+    Assert.assertTrue(dashboardRecords.stream()
+                        .filter(record -> SCHEDULED_PROG2_ID.getProgram().equals(record.getProgram()))
+                        .allMatch(schedule -> ArtifactSummary.from(ARTIFACT2_ID).equals(schedule.getArtifact())));
```

Comment:
ideally we need to check the actual objects for equality, this will also need implementing equals and hashCode in DashboardRunRecord class, i have added that in https://github.com/caskdata/cdap/pull/10254, just add a todo to update this test csse after that is merged.